joomla/joomla-cms
# [staging] [fix] Bugs in multilingual associations component

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/18552
(The original issue, plus the further discovered ones with `Publishing Options`)
Plus solving other bug.

### Summary of Changes
1. Adding hidden `id` field in layouts/joomla/edit/global.php
2. Moving code for the allowed categories to choose from in `Select Target` to the correct view
3. Solving https://github.com/joomla/joomla-cms/issues/18552 by preventing race condition caused by loading Target iframe at the same time as Reference. Thank you Robbie Jackson !!!


### Testing Instructions
Set a multingual sites with some articles associations and some categories set to All languages and different content languages.

Bug1. 
In Articles Global Options, set `Show Publishing Options` to hide

<img width="662" alt="screen shot 2018-09-13 at 15 33 56" src="https://user-images.githubusercontent.com/869724/45491680-8fb1a300-b76a-11e8-889e-326979a06e35.png">
Go to the multilingual component. Choose articles and a language. Try to load the sidebyside for an article ot its associated item. We get in a loop.


![jform_id](https://user-images.githubusercontent.com/869724/45492469-b244bb80-b76c-11e8-8a30-978be7480aa5.gif)

Bug2. 
Load the side by side view from clicking on an article or contact or Newsfeed (after setting `Show Publishing Options` back to Show.
In the Target part click on `Select Target` (or `Change Target` to load the modal.
In the modal click on SearchTools.
Then click on `Select a Category`
The dropdown will display all categories, although it should ONLY display categories tagged to ALL languages or to the language of the Target item.

<img width="760" alt="screen shot 2018-09-13 at 15 52 29" src="https://user-images.githubusercontent.com/869724/45493045-259afd00-b76e-11e8-8ea7-71601e847912.png">



Bug3. 
Filter by articles and one of the Content languages.
The items already associated show as a grey button.
Click on one of these grey buttons to load the side by side page with the already defined target (tagged to the language of the grey button)

The first time one clicks, the already associated target first appear for a while and then disappear to finally display a blank article.
Close the side by side page and then click again the same grey button.
This time all is fine, the already associated target displays fine.

![com_assoc](https://user-images.githubusercontent.com/869724/32688361-97a6bb04-c6cf-11e7-91e1-85ceb8b1ccf5.gif)

This may depend on the session and cache used on site. I get this all the time on my test sites


### After patch
Bug1 solved as we always have a `jform_id` obtained from the hidden id field

Bug2 solved

<img width="739" alt="screen shot 2018-09-13 at 15 40 37" src="https://user-images.githubusercontent.com/869724/45493337-c5f12180-b76e-11e8-85bf-9c8a94393943.png">


Bug3 solved. No need to reload.

### Documentation Changes Required
None
